### DISCUSSION TITLE
Port to Arch Linux ARM?

### DISCUSSION THREAD

**Discussion by mxmlkzdh** [pivpn/pivpn#1391] [Feature requests]
Any plans for porting this awesome project to Arch Linux ARM?

**Comment by 0-kaladin** [comment]
No plans... But maybe when I go to openvpn 2.4 I could. I do like arch.

**Comment by mxmlkzdh** [comment]
Yeah, that'd be great. I'd be more than happy to contribute (even an insignificant amount) to this awesome project. (I have some experience with Arch.) Do you happen to have anything in mind that I can start working on or looking into?

**Comment by 0-kaladin** [comment]
Not sure.  Really it'll be detecting the install is using Arch, then making the install changes to support openvpn.  Which might not be too different once i'm on openvpn 2.4 as I might be installing from a compiled src then as opposed to package maintainer.

**Comment by mxmlkzdh** [comment]
Sounds great! Thank you again.

**Comment by MikeRixWolfe** [comment]
I'd also love to see this patched to support installation and setup on arch-arm!

**Comment by Mausy5043** [comment]
Would it already fly on Arch Linux? Or is this completely Debian-dependent?  
I ask, because I have `pihole` running on Arch (non-ARM) and would like to also install `pivpn` on the same machine.

**Comment by coolapso** [comment]
@Mausy5043 nope, arch is still not supported at all.

**Comment by Drugantibus** [comment]
Is there any news on this? I think a lot of people are moving to Arch Linux ARM with the recent RPiOS/Microsoft fiasco and this project is almost a must have on a Pi imo. Even a "roadmap" with tasks needed to port this would be nice, so that people could start opening PRs :)

**Comment by coolapso** [comment]
@Drugantibus Most likely this will be dropped and the support will come in form of docker instead!

**Comment by coolapso** [comment]
OBS: Closed accidentaly

**Comment by Drugantibus** [comment]
That's even better for me! :)

**Comment by RobotsAreCrazy** [comment]
Any updates on this as wanting to try it out, installing arch-arm pi4?

**Comment by RobotsAreCrazy** [comment]
Found this: https://github.com/ljishen/pivpn-docker but no idea if it would work

**Comment by RobotsAreCrazy** [comment]
Also found this: https://hub.docker.com/r/linuxserver/wireguard

**Comment by coolapso** [comment]
There are no news.. and i am still to work on this ... as i have a bunch of concerns i want to endorse first ... 


1 - PiVPN is not statless, and Docker image with pivpn/openvpn or wireguard pre-installed and configured its a really bad idea because then there will be a lot of users with the same certificates / ciphers etc etc .. Reason why i don't recommend anyone to use the existing ones! 

2 - It has to run an unattended install on the entry point and detect for Persistent or custom settings and skip the configuration and cipher generation steps. 

So i'd say there still some work ahead ... don't expect this before summer, but i ensure you it will be done as i am really eager to get this done ... but Life gets in the way and at the moment i am working on other priorities.

**Comment by Offerel** [comment]
Currently I'm on migrating from Debian to Arch. Any new regarding the Arch support after the summer is gone?

**Comment by coolapso** [comment]
@Offerel all still as i've mentioned in the comments.

and I don't expect to be able to make a proper docker image before the end of this year unfortunately 

Pull requests are encouraged tho!

**Comment by Offerel** [comment]
Many thx. But one more question. If I have understood the concept correctly and I run the command `curl https://raw.githubusercontent.com/pivpn/pivpn/master/auto_install/install.sh | bash`, the script serves to install the necessary packages from the respective official repositories and then preconfigure these services. Quasi as simplification of a manual installation. Accordingly, at least in the case of Arch Linux, the appropriate packages from the official repositories should also be able to be installed. I could configure them manually if necessary. 

Or would the installation already fail with Arch?

**Comment by coolapso** [comment]
@Offerel your understanding of the concept is 100% correct, we only add a few scripts more to make it easier to generate client certificates but that's pretty much all about it. 

About arch, the installation will fail straight away for mainly for a couple of reasons: 

* The package manager is different, debian based distros use `apt` while arch uses `pacman`
* Openvpn package names actually don't change on OpenVPN scenarion, but for wireguard the case is already a bit different

then there's a bunch of other ones script internal related. 
as for configuration files  they should end up pretty much in the same places on both. 


This being said, the Pure Port itself on the script should not be that hard, but supporting all those OS also becomes hard to maintain code-wise and to make sure it works all the time. so ... as i stated before, i'd rather spend the time and make sure we can provide a proper and official docker image which will run in whatever OS you might want to use pivpn at. 

The problem on my side, even though i'd really love to work on that is mainly time!

**Comment by Offerel** [comment]
I completely understand your point. Take your time. Many thanks for your statement. I will try to setup the needed packages and configuration by myself with the help of the script. It doesn't matter if this will fail, since this is a  backup Raspberry Pi,  which I didn't use mainly.

**Comment by Da-Real-Kryall** [comment]
Any further news on this? I've recently moved my pi over to Manjaro Linux and it'd be great to get pivpn up and running again.

**Reply by coolapso** [reply]
There won't be any OS Specific ports ... we have added support for alpine recently so it can open the door for a docker image. 

An official docker image is still far from happening as it requires a lot of re-structuring on the code and also its not something particularly easy to setup the vpn protocols on docker either, proof of it is the fact that there are not many (if any) decent and safe docker images for any of the protocols.
